### ISSUE TITLE
Make the installation steps more user-friendly

### ISSUE THREAD

**Issue by anglixue** [raymondlouie/MiniMarS#5]
I have summarized several points I think we can improve the user expression when installing the pre-requested packages.

1. Shall we give instructions for pre-installed method packages? Like the following
```
# CiteFuse
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install("CiteFuse")

# sc2marker
if (!require("devtools", quietly = TRUE))
  install.packages("devtools")
devtools::install_github("https://github.com/CostaLab/sc2marker", build_vignettes = TRUE)

# geneBasisR
devtools::install_github("MarioniLab/geneBasisR") 
```

2. I remember on the readme page we have listed all four methods included in this package. What is the reason for removing it?

3. When loading the library, check the required packages
I can add those functions later

4. Add links with the required packages in the readme pages. So users can directly go to the original websites
https://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html
https://github.com/CostaLab/sc2marker
https://github.com/MarioniLab/geneBasisR
etc...

5. When I run the code with the first method, 'CiteFuse', the following error occurred. 

```
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘propr’
```

I confirmed the package is properly installed. I checked the website and 'propr' should be imported from the package. Has anyone met this before?

6. I've started a pull request where I add a function to check the user input method. Could anyone review it and merge into the main branch?
https://github.com/raymondlouie/ClusterMarkers/pull/4

7. At the end of the readme page, maybe print out your sessionInfo() so users will know about your development environment. Make sure you clear all your R environment and loaded packages before starting the whole pipeline, so you won't import some packages elsewhere unconsciously. 

8. I still have problems installing the package directly with the URL. This will be the first line of code that users will use. We might need to work it out to give a good first impression to users/reviewers. 
One reason could be this is still a private repository. After you make it public we could try again.

```
> devtools::install_github("raymondlouie/ClusterMarkers") 
Error: Failed to install 'unknown package' from GitHub:
  HTTP error 404.
  Not Found

  Did you spell the repo owner (`raymondlouie`) and repo name (`ClusterMarkers`) correctly?
  - If spelling is correct, check that you have the required permissions to access the repo.
```

We can discuss more in the meeting. Thank you！

**Comment by raymondlouie** [comment]
Thanks heaps Angli for the useful comments. I've should have implemented all of them, but please let me know if I forgot something.

**Comment by ghar1821** [comment]
Hi guys, just a suggestion. Rather than getting the users to install the dependencies themselves separately, you can perhaps change the DESCRIPTION file to automatically install those packages as dependencies. 

So add something like the following to the DESCRIPTION file: 

```
biocViews:
        CiteFuse
        SingleCellExperiment
Depends: R (>= 2.10)
Imports:
        Seurat
        xgboost
        dplyr
Remotes:
        tpq/propr
        CostaLab/sc2marker
        MarioniLab/geneBasisR
Suggests:
        devtools
```

Theoretically, the biocViews will automatically install packages from bioconductor, but I haven't test it myself.
